dll/win32/comctl32/rebar.c

```diff
@@ -2548,10 +2548,15 @@ REBAR_InsertBandT(REBAR_INFO *infoPtr, INT iIndex, const REBARBANDINFOW *lprbbi,
 
     /* initialize band */
     memset(lpBand, 0, sizeof(*lpBand));
+#ifdef __REACTOS__
+    lpBand->clrFore = infoPtr->clrText == CLR_NONE ? CLR_DEFAULT : infoPtr->clrText;
+    lpBand->clrBack = infoPtr->clrBk == CLR_NONE ? CLR_DEFAULT : infoPtr->clrBk;
```

Comment:
Weird, so, are the colors stored in `infoPtr->clrBtnText` and `clrBtnFace` correct for their intended purpose, though ?

Reply:
The existing drawing code is

```C
	if (lpBand->clrFore != CLR_NONE) {
	    new = (lpBand->clrFore == CLR_DEFAULT) ? infoPtr->clrBtnText :
		    lpBand->clrFore;
	    oldcolor = SetTextColor (hdc, new);
	}
```

infoPtr->clrBtnText gets updated with the syscolor on WM_SYSCOLORCHANGE (a specific ROS fix Wine does not have). Before my change, lpBand->clrFore gets set to clrBtnText (the color it is at that point in time) and is never updated.

Reply:
Ok